lightkurve/targetpixelfile.py

```diff
@@ -944,33 +991,19 @@ def extract_aperture_photometry(self, aperture_mask='pipeline'):
             Array containing the summed flux within the aperture for each
             cadence.
         """
-        aperture_mask = self._parse_aperture_mask(aperture_mask)
-        if aperture_mask.sum() == 0:
-            log.warning('Warning: aperture mask contains zero pixels.')
-        centroid_col, centroid_row = self.estimate_centroids(aperture_mask)
-        # Ignore warnings related to zero or negative errors
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore", RuntimeWarning)
-            flux_err = np.nansum(self.flux_err[:, aperture_mask]**2, axis=1)**0.5
-
-        keys = {'centroid_col': centroid_col,
-                'centroid_row': centroid_row,
-                'quality': self.quality,
-                'channel': self.channel,
-                'campaign': self.campaign,
+        lc = super().extract_aperture_photometry(aperture_mask=aperture_mask)
+        key_names = ['centroid_col','centroid_row','quality','ra','dec',
+                        'label','targetid']
+        keys = {key:getattr(lc, key) for key in key_names}
+        kepler_keys = {'campaign': self.campaign,
                 'quarter': self.quarter,
-                'mission': self.mission,
                 'cadenceno': self.cadenceno,
-                'ra': self.ra,
-                'dec': self.dec,
-                'label': self.header['OBJECT'],
-                'targetid': self.targetid}
-        return KeplerLightCurve(time=self.time,
-                                flux=np.nansum(self.flux[:, aperture_mask], axis=1),
-                                flux_err=flux_err,
+                'mission': self.mission,
+                'channel': self.channel}
+        keys.update(kepler_keys)
```

Comment:
I don't understand why we are using a `keys` dictionary here.  I'd get rid of `keys` altogether and, instead, simply pass the arguments directly via the constructor of `KeplerLightCurve` being called on the line below.